src/Dfe.PlanTech.Application/Extensions/QueryableExtensions.cs

```diff
@@ -0,0 +1,52 @@
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Dfe.PlanTech.Application.Extensions;
+
+/// <summary>
+/// Extension methods for caching database commands by QueryString
+/// These do not have any cache invalidation and should only be used on CMSDbContext queries, not PlanTechDbContext
+/// </summary>
+public static class QueryableExtensions
+{
+    private static MemoryCache _cache = new(new MemoryCacheOptions());
+    private const int CacheDurationMinutes = 30;
+
+    private static string GetCacheKey(IQueryable query)
+    {
+        var queryString = query.ToQueryString();
+        var hash = SHA256.HashData(Encoding.UTF8.GetBytes(queryString));
+        return Convert.ToBase64String(hash);
+    }
+
+    private static async Task<TResult> GetOrCreateAsyncWithCache<T, TResult>(
+        IQueryable<T> queryable,
+        Func<IQueryable<T>, CancellationToken, Task<TResult>> queryFunc,
+        CancellationToken cancellationToken = default)
+    {
+        var key = GetCacheKey(queryable);
+        return await _cache.GetOrCreateAsync(key, entry =>
+        {
+            entry.AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(CacheDurationMinutes);
```

Comment:
Do we need this? Isn't this part of the `MemoryCache`/`MemoryCacheOptions` inbuilt functionality?

Reply:
I couldn't find a way to set it globally in the options 🤔

Reply:
Oh yeah you're right. I was thinking of `MemoryCacheEntryOptions`. That's dumb